packages/loader/container-loader/src/loader.ts

```diff
@@ -275,10 +275,10 @@ export class Loader implements IHostLoader {
     private readonly logger: ITelemetryLogger;
 
     constructor(loaderProps: ILoaderProps) {
-        const scope = { ...loaderProps.scope };
-        if (loaderProps.options?.provideScopeLoader !== false) {
-            scope.ILoader = this;
-        }
+        const scope: FluidObject<ILoader> = {
+            ...loaderProps.scope,
+            ILoader: loaderProps.options?.provideScopeLoader !== false ? this : undefined,
```

Comment:
Probably prefer the conditional add still -- such that when `false`, `ILoader` is completely absent rather than present and `undefined`.  Also (though I don't think this is a scenario at present) in case someone wanted to pass their own `scope.ILoader` and not have it stomped on.